Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -199,4 +207,182 @@ theorem unifTight_finite [Finite ι] (hp_top : p ≠ ∞) {f : ι → α → β}
 
 end UnifTight
 
+section VitaliConvergence
+
+variable {μ : Measure α} {p : ℝ≥0∞} {f : ℕ → α → β} {g : α → β}
+
+/-! Both directions and an iff version of Vitali's convergence theorem on measure spaces
+of not necessarily finite volume. See `Thm III.6.15` of Dunford & Schwartz, Part I (1958). -/
+
+/- We start with the reverse direction. We only need to show that uniform tightness follows
+from convergence in Lp. Mathlib already has the analogous `unifIntegrable_of_tendsto_Lp`
+and `tendstoInMeasure_of_tendsto_snorm`. -/
+
+/-- Intermediate lemma for `unifTight_of_tendsto_Lp`. -/
+theorem unifTight_of_tendsto_Lp_zero (hp' : p ≠ ∞) (hf : ∀ n, Memℒp (f n) p μ)
+    (hf_tendsto : Tendsto (fun n => snorm (f n) p μ) atTop (𝓝 0)) : UnifTight f p μ := fun ε hε ↦by
+  rw [ENNReal.tendsto_atTop_zero] at hf_tendsto
+  obtain ⟨N, hNε⟩ := hf_tendsto ε (by simpa only [gt_iff_lt, ENNReal.coe_pos])
+  let F : Fin N → α → β := fun n => f n
+  have hF : ∀ n, Memℒp (F n) p μ := fun n => hf n
+  obtain ⟨s, hμs, hFε⟩ := unifTight_fin hp' hF hε
+  refine ⟨s, hμs, fun n => ?_⟩
+  by_cases hn : n < N
+  · exact hFε ⟨n, hn⟩
+  · exact (snorm_indicator_le _).trans (hNε n (not_lt.mp hn))
+
+/-- Convergence in Lp implies uniform tightness. -/
+private theorem unifTight_of_tendsto_Lp (hp' : p ≠ ∞) (hf : ∀ n, Memℒp (f n) p μ)
+    (hg : Memℒp g p μ) (hfg : Tendsto (fun n => snorm (f n - g) p μ) atTop (𝓝 0)) :
+    UnifTight f p μ := by
+  have : f = (fun _ => g) + fun n => f n - g := by ext1 n; simp
+  rw [this]
+  refine UnifTight.add ?_ ?_ (fun _ => hg.aestronglyMeasurable)
+      fun n => (hf n).1.sub hg.aestronglyMeasurable
+  · exact unifTight_const hp' hg
+  · exact unifTight_of_tendsto_Lp_zero hp' (fun n => (hf n).sub hg) hfg
+
+
+/- Next we deal with the forward direction. The `Memℒp` and `TendstoInMeasure` hypotheses
+are unwrapped and strengthened (by known lemmas) to also have the `StronglyMeasurable`
+and a.e. convergence hypotheses. The bulk of the proof is done under these stronger hypotheses.-/
+
+theorem tendsto_Lp_of_tendsto_ae_of_meas (hp : 1 ≤ p) (hp' : p ≠ ∞)
```

Comment:
This one is completely superseded by the next, right? If so, can you mark it private?